### PR TITLE
Add metadata columns to SOM cluster and consensus cluster data

### LINKED ISSUE #416: Add pixel metadata to the SOM cluster and consensus cluster data
**Is your feature request related to a problem? Please describe.**

Although the preprocessed data for SOM clustering contains the metadata and can easily be mapped, for ease of use it makes sense to include them in the SOM and consensus cluster data too.

**Describe the solution you'd like**

We read in the entire datasets for running the trained SOM and consensus clustering so we can access these metadata columns. The `markers` variables can be used for subsetting on channels for the actual computations.

### PR DESCRIPTION
**Pull request by alex-l-kong**
**What is the purpose of this PR?**

Addresses and closes #416. We add the metadata columns to both the SOM and consensus clustered data so it will be easy to access for users.

**How did you implement your changes**

We change the way we read in the `feather` dataframes from subsetting on just channel columns to reading in everything. We then use the `markers` vectors to subset the data for computational purposes, but we add additional columns to the full data.

### CODE REVIEW

File: ark/phenotyping/run_trained_som.R [hunk 1]
```diff
@@ -44,31 +44,32 @@ markers <- colnames(somWeights)
...
         }
     }
 
     # map FlowSOM data
-    clusters <- FlowSOM:::MapDataToCodes(somWeights, fovPixelData)
+    clusters <- FlowSOM:::MapDataToCodes(somWeights, as.matrix(fovPixelData[, markers]))
 
     # assign cluster labels column to pixel data
-    fovPixelData <- as.matrix(cbind(as.matrix(fovPixelData), cluster=clusters[,1]))
+    fovPixelData <- cbind(fovPixelData, cluster=clusters[,1])
```
Comment (ngreenwald) [inline ark/phenotyping/run_trained_som.R:63]: I think you said cbind was the slow part here, right? It is faster to initialize this matrix from the beginning with a blank consensus clustered column, and assign it here? Or is assignment equally slow

**Review by ngreenwald — CHANGES_REQUESTED** [review]
Looks good, minor comment

File: ark/phenotyping/run_trained_som.R (hunk 1 above)
Comment (alex-l-kong) [reply ark/phenotyping/run_trained_som.R:63]: Let me try using `$` assignment first; that's what Candace does in her scripts. Initializing this matrix with a blank column and assigning later would essentially be the same, just with an extra step.

Comment (ngreenwald) [reply ark/phenotyping/run_trained_som.R:63]: Sounds good. There's probably some best practices about fastest way to update dataframes

**Review by ngreenwald — APPROVED** [review]

**Pull request merged**